tokens/delivery/http/tokens_delivery.go

```diff
@@ -42,39 +67,125 @@ func NewTokensHandler(e *echo.Echo, ts mvc.TokensUsecase, logger log.Logger) {
 // @ID get-token-metadata
 // @Produce  json
-// @Param  denom  path  string  true  "Denom can either be a human denom or a chain denom"
-// @Success 200 {object} domain.Token  "Success"
-// @Router /tokens/metadata/{denom} [get]
-func (a *TokensHandler) GetMetadta(c echo.Context) (err error) {
+// @Param  denoms  path  string  false  "List of denoms where each can either be a human denom or a chain denom"
+// @Success 200 {object} map[string]domain.Token "Success"
+// @Router /tokens/metadata [get]
+func (a *TokensHandler) GetMetadata(c echo.Context) (err error) {
 	ctx := c.Request().Context()
 
-	denom := c.Param("denom")
+	denomsStr := c.QueryParam("denoms")
+	if len(denomsStr) == 0 {
+		tokenMetadata, err := a.TUsecase.GetFullTokenMetadata(ctx)
+		if err != nil {
+			return c.JSON(http.StatusInternalServerError, domain.ResponseError{Message: err.Error()})
+		}
+		return c.JSON(http.StatusOK, tokenMetadata)
+	}
+
+	denoms := strings.Split(denomsStr, ",")
+
+	tokenMetadataResult := make(map[string]domain.Token, len(denoms))
+
+	for _, denom := range denoms {
+		denom, err := url.PathUnescape(denom)
+		if err != nil {
+			return c.JSON(http.StatusBadRequest, domain.ResponseError{Message: err.Error()})
+		}
+
+		if err := sdk.ValidateDenom(denom); err != nil {
+			return c.JSON(http.StatusBadRequest, domain.ResponseError{Message: err.Error()})
+		}
+
+		tokenMetadata, err := a.TUsecase.GetMetadataByChainDenom(ctx, denom)
+		if err == nil {
+			return c.JSON(http.StatusOK, tokenMetadata)
+		}
+
+		// If we fail to get metadata by chain denom, assume we are given a human denom and try to translate it.
+		chainDenom, err := a.TUsecase.GetChainDenom(ctx, denom)
+		if err != nil {
+			return c.JSON(http.StatusInternalServerError, domain.ResponseError{Message: err.Error()})
+		}
+
+		// Repeat metadata retrieval
+		tokenMetadata, err = a.TUsecase.GetMetadataByChainDenom(ctx, chainDenom)
+		if err != nil {
+			return c.JSON(http.StatusInternalServerError, domain.ResponseError{Message: err.Error()})
+		}
 
-	denom, err = url.PathUnescape(denom)
+		tokenMetadataResult[chainDenom] = tokenMetadata
+	}
+
+	return c.JSON(http.StatusOK, tokenMetadataResult)
+}
+
+// @Summary Get prices
+// @Description Given a list of base denominations, returns the spot price with a system-configured quote denomination.
+// @Accept  json
+// @Produce  json
+// @Param   base          query     string  true  "Comma-separated list of base denominations (human-readable or chain format based on humanDenoms parameter)"
+// @Param   humanDenoms   query     bool    false "Specify true if input denominations are in human-readable format; defaults to false"
+// @Success 200 {object} map[string]map[string]string "A map where each key is a base denomination (on-chain format), containing another map with a key as the quote denomination (on-chain format) and the value as the spot price."
+// @Router /tokens/prices [get]
+func (a *TokensHandler) GetPrices(c echo.Context) (err error) {
```

Comment:
Operationalize /tokens/prices endpoint 

- Load test
- Fuzz unit test with various combinations of mainnet denoms from mainnet state
- Add instrumentation
- Build dashboards